### ISSUE TITLE
False "Multiple exports of name" error when two "export * as ..." contain same name [import/export] 

### ISSUE THREAD

**Issue by Stijnkool** [import-js/eslint-plugin-import#2289]
The issue I am having seems related to https://github.com/import-js/eslint-plugin-import/issues/1834
That specific issue is fixed. But I am having the same issue when using an extra level of indirection.
TypeScript does not seem to have a problem with the added level. So I guess this plugin should allow it...?

## Scenario described in https://github.com/import-js/eslint-plugin-import/issues/1834
./A/A1.ts
```ts
export type X = "A1";
```
./A/A2.ts
```ts
export type X = "A2";
```
./A/index.ts
```ts
export * as A1 from "./A1";
export * as A2 from "./A2";
```
In this scenario, as described in https://github.com/import-js/eslint-plugin-import/issues/1834 there would be an error `Multiple exports of name 'X'`.
There is no error. This is fixed.

## My scenario
I have the same files as above. With the following added:
./B/B.ts
```ts
export type X = "B";
```
./B/index.ts
```ts
export * as B from "./B";
```
./index.ts
```ts
export * as A from "./A";
```

**Comment by ljharb** [comment]
Yes, I agree this should work.

**Comment by darrenvong** [comment]
Not sure how much this helps, but this issue was fixed for me from `2.25.2` up to `2.28.2` using eslint 8. `2.29.0` onwards is where I'm seeing the same issue again

**Comment by ljharb** [comment]
Fixed by #3250, now on `main` (ships in the next release). The false `import/export` "Multiple exports of name" error for two `export * as` re-exports sharing a name no longer reproduces with that change.
